src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react";
+
+import { postComment } from "../../api";
+
+const CommentAdder = ({ article_id, setComments }) => {
+  const [newComment, setNewComment] = useState("");
+  const [newUser, setNewUser] = useState("jessjelly");
+  const [err, setErr] = useState(null);
+  const [validateComment, setValidateComment] = useState(true);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    e.target.disabled = true;
+    console.log("button clicked");
+    if (newUser && newComment) {
+      postComment(article_id, { username: newUser, body: newComment })
+        .then(() => {
+          setComments((currComments) => {
+            return [
+              { comment_id: 9999, author: newUser, body: newComment, votes: 0 },
```

Comment:
It would be better if you used the returning object from the post request to update the list. 

This will work for now, but you'll find you can't delete the comment without refreshing it, because it wont have the correct db. 

I'd get the article that the post returns for you on line 17, then use that instead of your comment_id object you've made.